### DISCUSSION TITLE
Expression evaluation

### DISCUSSION THREAD

**Discussion by djpirra** [duckdb/duckdb#12648] [Q&A]
Hi everyone,

I have a configuration table that has a data column where the value is a SQL expression that should be evaluated when executed in another query.

For example lets say the config table is a query generator table that has a source column and an expression column.

In the expression column you might have values such as [A] / [B].
Now you want to perform this processing while running a join with this table and the text of the expression above should be evaluated and returned as a value.

Is it possible to achieve this in DuckDB?

In SQL Server back in the day I built a .Net extension for this.

Thank you everyone

**Comment by tonne1** [comment]
Hi,

there is no dynamic query evaluation yet in duckdb AFAIK. There is one mechanism,  
```sql
SELECT * FROM json_execute_serialized_sql(json_serialize_sql('SELECT 1 + 2'));
```
which comes close to an 'execute statement' within duckdb, but also this function doesn't accept lateral parameters, in this case the sql text, so this won't help you.

In the CLI you can 

```sql
-- store the results of the query in cmds.sql
.mode column
.headers off
.once 'cmds.sql'

-- construct your query by running your statement

-- execute result
.read 'cmds.sql'
```

But that's a round trip over the file system.
